### DISCUSSION TITLE
Use filters for bulk update

### DISCUSSION THREAD

**Discussion by jaulz** [LastDragon-ru/php-packages#178] [Q&A] [answered]
Sorry, it's me again ☺️ I would like to implement a bulk update based on a query with certain filters. Is there any way how I can reuse the filters that I created for the GraphQL query and use them as input for a mutation? In the mutation I would need to instantiate the query builder with the filters from the input then.

Thanks for any help ✌️

**Comment by LastDragon-ru** [comment]
To be honest, I've never tried it. Probably if you add `@searchBy`/`@sortBy` arg to any mutation the same as in query (maybe [explicit type](https://github.com/LastDragon-ru/lara-asp/blob/main/packages/graphql/README.md#input-type-auto-generation) will be needed) the types should be created in the schema. And then you can use `ResolveInfo::enhanceBuilder()` inside mutation to apply these args to the builder.

**Reply by jaulz** [reply]
Nice, thanks I will give it a try 😊  Could you also give me please a hint where I can find how the filters are applied to the query builder in your library?

**Reply by LastDragon-ru** [reply, marked as answer]
Both directives just implements `Nuwave\Lighthouse\Support\Contracts\ArgBuilderDirective` and `Nuwave\Lighthouse\Scout\ScoutBuilderDirective`. The `ResolveInfo::enhanceBuilder()` doing all other stuff. See https://lighthouse-php.com/6/custom-directives/input-value-directives.html#argbuilderdirective

If you have access to `Argument`/`ArgumentSet` you can try something like:

https://github.com/LastDragon-ru/lara-asp/blob/a25feab2d7fcb0fd3aded206aa26784ae1b94ca1/packages/graphql/src/Stream/Directives/Directive.php#L492-L513

**Reply by jaulz** [reply]
@LastDragon-ru thanks a lot for your help! I got it working :)
